### DISCUSSION TITLE
Feedback on PyFluent with other Python ecosystem

### DISCUSSION THREAD

**Discussion by PySRG** [ansys/pyfluent#74] [Show and tell]
Hello,

I have been exploring PyFluent with other Python ecosystem such as Numpy, Pandas, scikit-learn, Seaborn, Plotly,... I am able to create an example which performs DOE, fits ML models and predicts the output for unseen data. Overall experience is very positive! 

Following are key highlights:

- All packages are installed under a separate conda environment. 
- **Jupyter Notebook** is the only IDE used so far. It worked as expected. Auto-Complete worked nicely for PyFluent.
- **PyFluent** usage was limited to a few APIs. Case setup (Mixing Elbow) was done a priori in a standard Fluent.
- **DOE:** Defined manually using **Numpy** arrays for Cold and Hot Velocity Inlet. Average temperature at outlet was extracted as an output using TUI API, stored in a 2D Numpy array.
-  **Response Surface:** Plotted using **Plotly**. Faced some error initially but was resolved after installing jupyter, ipykernel and nbformat in this separate conda env.
- **Data Pre-Processing:** Using **Pandas** and **scikit-learn**.
- **ML Model Fitting and Prediction:** **Linear Regression** with **Polynomial features** and **Random Forest Regressor** of **scikit-learn**. **XGBoost Regressor** installed as a separate library.
- **Parity Plot**: **Seaborn** and **Matplotlib**.

Regards,
Shitanshu.

**Comment by dnwillia-work** [comment]
@PySRG Awesome!  Please let us know if anything you require is missing or not working.

**Comment by donff2j** [comment]
This is fantastic. I would like to see a demo and overview.

**Comment by PySRG** [comment]
@donff2j @dnwillia-work

Please see a short demo here: 

https://user-images.githubusercontent.com/99811447/156543663-719912ab-495a-41b2-ae13-39a70761829a.mp4

Regards,
Shitanshu.
